src/main/java/org/jenkinsci/plugins/tuleap_credentials/AccessKeyChecker.java

```diff
@@ -0,0 +1,42 @@
+package org.jenkinsci.plugins.tuleap_credentials;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import org.jenkinsci.plugins.tuleap_api.AccessKeyApi;
+import org.jenkinsci.plugins.tuleap_api.AccessKeyScope;
+import org.jenkinsci.plugins.tuleap_credentials.exceptions.InvalidAccessKeyException;
+import org.jenkinsci.plugins.tuleap_credentials.exceptions.InvalidScopesForAccessKeyException;
+
+public class AccessKeyChecker {
+    private static ImmutableList<String> MANDATORY_SCOPES = ImmutableList.of("write:rest", "write:git_repository");
+
+    private AccessKeyApi client;
+
+    @Inject
+    public AccessKeyChecker(AccessKeyApi client) {
+        this.client = client;
+    }
+
+    public void verifyAccessKey(String accessKey) throws InvalidAccessKeyException, InvalidScopesForAccessKeyException {
```

Comment:
Would not it be better to pass a `Secret` here instead of the raw string? That would limit exposure in potential stack traces.

Reply:
Done